SQL_Assignments/sql_4_b.txt

```diff
@@ -0,0 +1,49 @@
+-- Questions
+
+-- 1. List all products with their category names, including products without a category.
+    select Productname,CategoryName from products p left join categories c on p.categoryid=c.categoryid
+
+-- 2. Display all customers and their order history, including customers who haven't placed any orders.
+
+    select customername,productname from Customers c left join orders o on c.CustomerID = o.CustomerID left join orderdetails od on od.orderid=o.orderid left join products p on od.productid = p.productid
+
+
+-- 3. Show all categories and the products in each category, including categories without any products.
+
+    select categoryname,productname from categories c left join products p on c.categoryid=p.categoryid 
+
+
+-- 4. List all possible customer-product combinations, regardless of whether a purchase has occurred.
+
+    select customername,productname from customers cross join products
+
+
+-- 5. Display all orders with customer and product information, including orders where either the customer or product information is missing.
+
+    select customername,productname from customers c join orders o on c.customerid = o.customerid join orderdetails od on o.orderid = od.orderid join products p on od.productid = p.productid
+
+
+-- 6. Show all products that have never been ordered, along with their category information.
+
+    select productname,categoryname from Products p left join OrderDetails od on p.ProductID = od.ProductID join Categories c on p.CategoryID = c.CategoryID where od.OrderDetailID is null
+
+
+-- 7. List all customers who have placed orders in the last week, along with the products they've purchased.
+
+    select customername,productname from Customers c join Orders o on o.CustomerID = c.CustomerID join OrderDetails od on o.OrderID = od.OrderID join Products p on od.ProductID = p.ProductID
```

Comment:
Is this correct ?

Reply:
select customername,productname from Customers c join Orders o on o.CustomerID = c.CustomerID join OrderDetails od on o.OrderID = od.OrderID join Products p on od.ProductID = p.ProductID where datepart(week,orderdate)=datepart(week,getdate())-1